lib/log_parser.rb

```diff
@@ -10,4 +13,18 @@ def first_line
     @file.close
     first_line
   end
+  
+  def generate_json
+    count_lines
+    hash_json = { "#{@log_file}": { "lines": @lines } }
+    JSON.pretty_generate(hash_json)
+  end
+
+  private
+
+  def count_lines
+    @lines = IO.readlines(@file).size
```

Comment:
@jesrodri não há necessidade em declarar o resultado daqui a uma variável de instância.
```suggestion
    IO.readlines(@file).size
```